src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclaratorHandler.java

```diff
@@ -381,15 +397,16 @@ private void processMembers(
           this.lang.getScopeManager().replaceNode(method, declaration);
         }
       } else if (declaration instanceof VariableDeclaration) {
-        FieldDeclaration fieldDeclaration =
-            FieldDeclaration.from((VariableDeclaration) declaration);
-        recordDeclaration.getFields().add(fieldDeclaration);
-        this.lang.replaceDeclarationInExpression(fieldDeclaration, declaration);
-
+        // recordDeclaration.getFields().add();
+        lang.getScopeManager()
+            .addDeclaration(FieldDeclaration.from((VariableDeclaration) declaration));
+        lang.getScopeManager().removeDeclaration(declaration);
       } else if (declaration instanceof FieldDeclaration) {
-        recordDeclaration.getFields().add((FieldDeclaration) declaration);
+        // recordDeclaration.getFields().add((FieldDeclaration) declaration);
+        lang.getScopeManager().addDeclaration(declaration);
       } else if (declaration instanceof RecordDeclaration) {
-        recordDeclaration.getRecords().add((RecordDeclaration) declaration);
+        // record is not stored as reference in the scope
+        lang.getScopeManager().addDeclaration(declaration);
```

Comment:
Same question as in previous PRs: Do you need the commented-out lines or can we remove them?

Reply:
Remove them, these were unintentionally not removed.